docs/adr/0035-refactor-payments-frontend.md

```diff
@@ -0,0 +1,101 @@
+Regarding why Next.js is favored over Gatsby.js, before completing the ADR, the subscription platform team went through a prototyping project and found the two frameworks to be remarkably similar in capabilities. However due to Next.js's popularity, use by other teams in Mozilla and a team vote, the decision was made to pick Next.js.
+
+## Pros and Cons of the Options
+
+### A. React Framework - Next.js
+
+Use a React framework, Next.js.
+
+- Good, because it provides many benefits out of the box, without compromising flexibility, that would require additional libraries otherwise. (Routing, rendering options, image optimization, etc.)
+- Good, because of its routing simplicity and flexibility
+- Good, because of its built in tooling, which reduces existing bespoke tooling currently used
+- Good, because it is easy to learn the basics
+- Good, because it's parent company has other tools that work well with Next.js and enhance its capabilities. (Turbopack, SWR, etc.)
+- Good, because of its large community support and popularity
+- Good, because it is used by other teams in Mozilla
+- Good, because it no longer uses Redux for state management
+- Bad, because it adds another layer on top of React
+- Bad, because it's new to most team members, and requires some learning to use more complicated features
+- Bad, because it diverges from tooling used elsewhere in the FxA monorepo
+
+### B. React Framework - Gatsby.js
+
+Use a React framework, Gatsby.js. (Gatsby has similar good and bad points to Next.js)
+
+- Good, because it provides many benefits out of the box, without compromising flexibility, that would require additional libraries otherwise. (Routing, rendering options, image optimization, etc.)
+- Good, because of its routing simplicity and flexibility
+- Good, because of its built in tooling, which reduces existing bespoke tooling currently used
+- Good, because it is easy to learn the basics
+- Good, because it no longer uses Redux for state management
+- Neutral, because of its large community support and popularity, but smaller compared to Next.js
+- Neutral, not currently used by other teams in Mozilla
+- Bad, because it adds another layer on top of React
+- Bad, because it's new to most team members, and requires some learning to use more complicated features
+- Bad, because it diverges from tooling used elsewhere in the FxA monorepo
+
+### C. React, Typescript, without Redux
+
+Use the existing fxa-payments-server app and only remove redux.
+
+- Good, because simplifies the app by removing complexity of using Redux
+- Good, because it would mostly use the existing technologies, and not require new learnings for the team
+- Bad, because Rescripts, CRA, and other tooling make it difficult to upgrade React tooling. (See [FXA-5242](https://mozilla-hub.atlassian.net/browse/FXA-5242))
+- Bad, because it will still require significant refactoring to remove Redux and introduce GraphQL
+- Bad, because page routing with react-router is complicated
```

Comment:
I personally don't find react-router that complex, but maybe SubPlat's experience was different?

Reply:
That's valid, that's been my experience with `react-router` as well. In this case it is just another library for folks to familiarize themselves with vs just copy-pasting a file. Which is something we are thinking of potentially doing, by allowing RPs to copy a template page and configure it themselves.

Reply:
> Would it be OK to reword this to the following?
> 
> `Bad, because page routing with react-router is more complicated, needing to learn a routing library, compared to the file-based routing used by popular React frameworks.`

Thanks for clarifying -- can you link to `react-router` and an example of this "file-based routing" as part of the sentence? That'd probably help folks like me understand the distinction a bit better.

Ah you know I just read up a bit on `react-router`, as it's been a while. It seems like something we'd use if we were keeping the SPA approach on the front-end, but it sounds like we want to go with a multi-page approach instead with actual network requests in between and browser navigation instead of updating the browsing history and URL to get some of the same browser features (i.e. clicking Back and Forward). Maybe you can reference something about that in the sentence too?